### ISSUE TITLE
 app.json is silently ignored when app.config.ts is present

### ISSUE THREAD

**Issue by lukewlms** [expo/expo#22706]
### Summary

I ran into multiple frustrating issues while setting up Expo for the first time. My config data was being completely ignored. It seems that the issue was that app.json and app.config.ts both existed. For example, my Android Version on EAS build came out as "1" instead of using the `android.version` field; and my iOS build was rejected by Apple because tablet support was missing, forcing me to use Xcode to create the build, not knowing why Expo was ignoring the Tablet config I'd put in `app.json`. These effects are pretty deleterious and make it look like Expo builds are broken.

It is fairly obscured in the docs that these files all serve the same purpose and only one should exist. At least, I think that is the case - after reading about Expo Config for a while I'm only fairly sure. It's just not explicit. (I think in my initial setup, two different config files were automatically created, which if true seems it would be its own bug.)

Other developers also have the same confusion about these files, e.g. [stackoverflow.com/questions/76109820/in-expo-what-is-the-difference-between-the-app-json-and-app-config-json](https://stackoverflow.com/questions/76109820/in-expo-what-is-the-difference-between-the-app-json-and-app-config-json)

If multiple of these files are included, Expo should simply warn the user on start that some of their config data is being ignored, and only one of those files is necessary. That would have avoided at least a couple hours of headaches in my case and also made me realize I could have confidence in EAS builds not ignoring my config.  E.g.:

> *** Warning: multiple config files were detected. Only one config file is needed.  Reading config from app.config.ts. app.json was also detected, and is being ignored.

I'm logging this as a bug because Expo docs many times say to update app.json, but doing that has no effect when an overriding config file also exists.

### What platform(s) does this occur on?

Android, iOS

### SDK Version

~48.0.11

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.3.1
      Shell: 5.5.1 - /usr/local/bin/zsh
    Binaries:
      Node: 16.20.0 - /usr/local/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 8.19.4 - /usr/local/bin/npm
      Watchman: 2022.12.12.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.12.0 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.10121639
      Xcode: 14.3/14E222b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~48.0.11 => 48.0.11
      react: 18.2.0 => 18.2.0
      react-native: 0.71.8 => 0.71.8
    npmGlobalPackages:
      eas-cli: 3.13.2
      expo-cli: 6.3.2
    Expo Workflow: bare

### Minimal reproducible example

Both in root directory:
- app.json
- app.config.ts

Result: app.json is totally ignored.

(I'm not sure how to create a better example than that since Snacks appear not to include app.json.)

**Comment by brentvatne** [comment]
hi there! this is intended behavior, but i do think we could do more to let people know if they forget to extend the app.json config in their app.config.js. so, for example:

**app.json**
```json
{
  "expo": {
     "name": "My App"
   }
}
```

**app.config.js**
```
export default (config) => ({
  slug: "my-app"
});
```

we could add a hidden property to the `config` object passed in to **app.config.js** and, if it's not found on the resulting evaluated config, then we can warn and explain how these values are related and suggest a fix.

how does this sound to you?

**Comment by lukewlms** [comment]
@brentvatne Yes, I think that would be a great and simple solution.  Could even be something like a field called "staticConfigSource" that passes "app.json" or similar.  

Just want to try and make sure others don't have the same confusion.  I also created pull requests for the Env Var docs to suggest using a function since I think most people will already have an `app.json`.

**Comment by EvanBacon** [comment]
We've added this as a check to `npx expo-doctor` the check itself is a bit expensive to perform on each config read so we can't do it automatically in an efficient way.
